Feat/trello integration

Ronaldo93 opened the pull request:
# Description

This introduces using Trello for task managements.

I have manually tested this (alongside automated API test) and so far it can retrieve the card from trello and added it. Current implementation has minor visual bugs here and there. Automation and ease of task creation should be next, hence the draft PR status.

## In-scope
- Create tasks from trello board.

## How to test it
Since this is still in early stage, much of the process linking still needs to be done manually. After doing the following steps, you should be able to put a task into inbox:

1. Create a provider and select trello.The following guides demonstrate how to get necessary detail for configuring Trello.
2. Create/Login Trello account.
3. Follow this [guide](https://developer.atlassian.com/cloud/trello/guides/rest-api/api-introduction/) to retrieve API key and the tokens.
4. Create a board (usually not needed since trello created for you a new one when you registration).
5. In application, configure Trello provider with token and API key.
6. After saving, select a board for adding related card.

Currently, the only way to add trello card is to done it via searching (I don't know if this is the case with other providers).

## Known limitations
- I am aware of the test not being ran correctly. However, it is because of at the time I pushing these changes, the test in pre-hook commit prevented me from pushing without any meaningful output, so I pushed with --no-verify and let CI/CD ran it instead.
- I haven't tested it via creating desktop application yet.
- Checklist within trello cards will *not* be worked on within this issue's scope, although future work on this is possible.
- Trello icons overlapping the drag handler (I don't know if this is by design).

## Issues Resolved
Fix #293. 

## Check List
I will update here once I have more time. Since its still actively being worked on, the following checklist was not selected.

- [x] New functionality includes testing.
- [x] New functionality has been documented in the README.
- [x] Remove the text requiring to use mobile/desktop or extension to use Trello Integration.
- [x] multiple search result.
- [x] add visual guide for configuring API keys and token.

Ronaldo93 committed:
feat(ui): add (non-working) trello provider button to the providers list

Ronaldo93 committed:
feat(trello_integration): non-functional stub for
trello configuration.

Ronaldo93 committed:
Merge branch 'master' into feat/trello-integration

Ronaldo93 committed:
fix: lint error.

Ronaldo93 committed:
feat(trello): wip - integrate trello with super productivity

Ronaldo93 committed:
feat(trello): initial working on trello integration.

This is still pretty much WIP; things may break and it still needs to do several thing manually, so I need your feedback

I have tested this and so far it can retrieve the card from trello and added it. Minor visual bugs here and there is seen, so that is one aspect. Automation and ease of task creation should be next.

johannesjo commented:
Thank you very much!  Looks very good so far! Let me know when you think this is ready for merge.

Ronaldo93 committed:
feat: select trello board after valid trello configuration.

Ronaldo93 commented:
For the trello app it needs to be created under a developer. I am leaning towards letting user creating it manually by themselves to ensure privacy, while we only provide way to do so. @johannesjo what do you think on this?

Ronaldo93 committed:
feat: Update README to include Trello in integrations

johannesjo commented:
> For the trello app it needs to be created under a developer. I am leaning towards letting user creating it manually by themselves to ensure privacy, while we only provide way to do so. @johannesjo what do you think on this?

Yes, I agree on this! Thanks for asking! It also makes sense, because should strive to keep the ongoing maintanance effort low and this is another thing that can change break and so on.

Ronaldo93 committed:
feat: added indication for boards loading.

* chore: added comment for clarification

Ronaldo93 committed:
feat(trello): removed warning from the dialog

* feat: reworded description (to be added later)
* feat(trellocfg): added label for better clarity.

Ronaldo93 committed:
feat(trello): removed validation for boardId

with trello board searching, user don't need to type in anymore, thus validation is not needed anymore.

Ronaldo93 committed:
Merge remote-tracking branch 'refs/remotes/origin/feat/trello-integration' into feat/trello-integration

Ronaldo93 committed:
fix: invalid test

Ronaldo93 committed:
feat: initial instruction on trello integration setup.

Ronaldo93 commented:
@johannesjo can you review the initial trello integration? Although it looks okay on my side, there could be 
some area that I might overlook.

Ronaldo93 committed:
fix: test case for trello

Ronaldo93 committed:
fix: removed hard check on the trello board id format.

Ronaldo93 committed:
feat: update guide.

Ronaldo93 commented:
@johannesjo it looks good on my end, maybe give it another try?

Ronaldo93 commented:
@johannesjo did you have something like this before getting token?

(image taken from the guide)
<img width="675" height="1304" alt="image" src="https://github.com/user-attachments/assets/c224ce36-f085-475b-ae3a-36341f2ed019" />

Maybe my written guide is incorrect, if that is the case let me know so that i can rework on it!

Ronaldo93 committed:
feat: additional button for loading

* feat(ux): updated instruction for loading trello board

Ronaldo93 committed:
fix: failed test

* feat: add icon in providers list

Ronaldo93 committed:
chore: adjusted text

johannesjo committed:
Merge branch 'master' into feat/trello-integration

jaysonwcs commented:
It's not clear what this integration is able to do. Can it sync the other way around (sync comments and status changed on Super Productivity to Trello)? Can it download other info from Trello cards beyond the title and members?
